lib/src/intl_suggestors/intl_message_syntax.dart

```diff
@@ -142,6 +142,22 @@ class MessageSyntax {
   String nameForNode(StringLiteral body,
       {String? initialName, bool startAtZero = false}) {
     var messageText = body.toSource();
+    //Strings from constant fields will not pass below condition so initialName will be unchanged.
+    // Needing condition initialName.endsWith('_intlFunction') bcz when any node has addTestId we need to give that
+    // priority, and will not override the node name.
+    if (body is StringInterpolation && (initialName == null || initialName.endsWith('_intlFunction'))) {
+
+      var strings = body.elements.where((each) => each is InterpolationString).map((each) => (each as InterpolationString).value).toList();
+      var data=strings.join(' ').trim();
+      if(data.isNotEmpty && !data.contains("'")){
+        String name = toVariableName(data);
+        if(name.trim().isNotEmpty){
+          var functionName = owner.nameForString(name, messageText,
+              startAtZero: false);
+          return functionName;
+        }
+      }
+    }
```

Comment:
Also, for failure cases, if 'data' is empty, then we're falling through to the default case for an interpolation. That should probably be more explicit. I think it probably works right now because in those cases initialName is never null, so the functionName from the interpolation is unused.

Reply:
Updated this failure case also.